src/burnchains/bitcoin/network.rs

```diff
@@ -70,6 +70,8 @@ impl BitcoinIndexer {
         let magic = network_id_to_bytes(self.runtime.network_id);
 
         self.with_socket(|ref mut sock| {
+            // abort faster!
+            sock.set_read_timeout(Some(std::time::Duration::from_secs(5)));
```

Comment:
Can you do this in `src/burnchains/bitcoin/indexer.rs`'s `reconnect_peer()` method?

Also, I'm not sure this truly solves the problem.  If the bitcoin node is under load (e.g. takes more than 5 seconds to reply), then this change can cause a node to stall forever.  I remember having to _increase_ the timeout in the Neon node at some point because I couldn't get my local follower to synchronize with the Neon bitcoin node.  Maybe the timeout should be configurable?